### PR TITLE
Additional negative tests

### PR DESCRIPTION
**Pull request by pnwamk**
See https://github.com/GaloisInc/argo/issues/31

### CODE REVIEW

File: file-echo-api/README.rst [hunk 1]
```diff
@@ -20,6 +20,11 @@ Commands
   * Clears any loaded file from the server's memory.
   * No parameters
 
+* ``implode``
```
Comment (david-christiansen) [inline file-echo-api/README.rst:23]: Excellent name :)

File: file-echo-api/file-echo-api/Main.hs [hunk 2]
```diff
@@ -29,5 +29,6 @@ serverMethods :: [(Text, Argo.MethodType, JSON.Value -> Argo.Method FES.ServerSt
 serverMethods =
   [ ("load",           Argo.Command,   Argo.method FES.loadCmd)
   , ("clear",          Argo.Command,   Argo.method FES.clearCmd)
+  , ("implode",        Argo.Command,   Argo.method FES.implodeCmd)
```
Comment (david-christiansen) [inline file-echo-api/file-echo-api/Main.hs:32]: Why is this a command, if the server never enters a new state through implosion? (yes, nit-picky)

File: file-echo-api/test-scripts/file-echo-tests.py [hunk 3]
```diff
@@ -95,6 +96,35 @@ def run_tests(c):
...
+    uid = c.send_message("show", {"state": "12345678-9101-1121-3141-516171819202"})
+    actual = c.wait_for_reply_to(uid)
+    expected = {'error':{'data':{'stdout':'','data':'12345678-9101-1121-3141-516171819202','stderr':''},'code':20,'message':'Unknown state ID'},'jsonrpc':'2.0','id':uid}
+    assert(actual == expected)
+
+    # invalid request (missing JSON-RPC required fields)
+    invalid_request = {'jsonrpc': '2.0','method': 'bad request', 'id':1}
+    c.process.send_one_message(json.dumps(invalid_request))
+    actual = c.wait_for_reply_to(None)
+    expected = {'error':{'data':{'stdout':'','data':'Error in $: key \"params\" not found','stderr':''},'code':-32700,'message':'Parse error'},'jsonrpc':'2.0','id':None}
```
Comment (david-christiansen) [inline file-echo-api/test-scripts/file-echo-tests.py:116]: I worry that we're tying this too closely to the precise text of the Aeson-generated error message. In this case, I'd instead assert that the code and message fields were what's expected.

**Review by david-christiansen — COMMENTED** [review]
I'm a bit concerned about hard-coding some internal auto-generated error message texts. But perhaps that is fine, and we'll just update the tests as it goes. What do you think?

File: file-echo-api/test-scripts/file-echo-tests.py (hunk 3 above)
Comment (pnwamk) [reply file-echo-api/test-scripts/file-echo-tests.py:116]: I don't entirely disagree.

Though, in this case would the "precise text of the Aeson-generated error message" here be something a user would potentially encounter? If so... it _might_ be nice to explicitly track/acknowledge it in some way and know when we change it (for better or worse).

Comment (pnwamk) [reply file-echo-api/test-scripts/file-echo-tests.py:116]: > I'm a bit concerned about hard-coding some internal auto-generated error message texts. But perhaps that is fine, and we'll just update the tests as it goes. What do you think?

I think I'm inclined to want _some_ testing of error message texts that users might see, but limit it to as few such tests as necessary to reasonably track that particular aspect of the human-computer interaction. I totally agree if we had many such tests then that would mean our testing was overly repetitive and brittle, which would just be a pain in the long run and not be helpful.

Comment (pnwamk) [reply file-echo-api/test-scripts/file-echo-tests.py:116]: After a chat, we decided to leave the test as is, with the thought we may want to improve these error messages slightly in the future (remove the `$:` meaning "top-level parsing object" for example)

**Pull request merged**